flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/sink/TestPartitionUdf.java

```diff
@@ -0,0 +1,224 @@
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testBucketTime() {
+    Transform<Object, Integer> bucket = Transforms.bucket(Types.TimeType.get(), 4);
+
+    String str = "14:08:59";
+    Integer obj = bucket.apply(Literal.of(str).to(Types.TimeType.get()).value());
+    String expectHumanString = bucket.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT buckets(4, TIME '%s')", str);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testBucketByteBuffer() {
+    Transform<Object, Integer> bucket = Transforms.bucket(Types.BinaryType.get(), 4);
+
+    Integer obj = bucket.apply(ByteBuffer.wrap(new byte[] {1, 2, 3, 4, 5, 6}));
+    String expectHumanString = bucket.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT buckets(4, x'010203040506')");
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testTruncateInt() {
+    Transform<Object, Object> truncate = Transforms.truncate(Types.IntegerType.get(), 4);
+
+    int num = 10;
+    Object obj = truncate.apply(num);
+    String expectHumanString = truncate.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT truncates(4, %d)", num);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testTruncateLong() {
+    Transform<Object, Object> truncate = Transforms.truncate(Types.LongType.get(), 4);
+
+    long num = 10;
+    Object obj = truncate.apply(num);
+    String expectHumanString = truncate.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT truncates(4, %d)", num);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testTruncateString() {
+    Transform<Object, Object> truncate = Transforms.truncate(Types.StringType.get(), 4);
+
+    String str = "abcdef";
+    Object obj = truncate.apply(str);
+    String expectHumanString = truncate.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT truncates(4, '%s')", str);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testTruncateDecimalType() {
+    Transform<Object, Object> truncate = Transforms.truncate(Types.DecimalType.of(6, 5), 4);
+
+    Object obj = truncate.apply(BigDecimal.valueOf(6.12345));
+    String expectHumanString = truncate.toHumanString(obj);
+
+    List<Row> sql = sql("select truncates(4, cast(6.12345 as decimal(6,5)))");
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testTruncateByteBuffer() {
+    Transform<Object, Object> truncate = Transforms.truncate(Types.BinaryType.get(), 4);
+
+    Object obj = truncate.apply(ByteBuffer.wrap(new byte[] {1, 2, 3, 4, 5, 6}));
+    String expectHumanString = truncate.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT truncates(4, x'010203040506')");
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
```

Comment:
I’m still reviewing these tests, but for any types that are _not_ supported, can we add a test checking what happens if those types are passed in?

Assuming an exception results, then `AssertHelpers` has some methods that should be useful.

Reply:
OK, I will add test cases for different types.